### PR TITLE
Feature/jobs db footer 2

### PR DESCRIPTION
**Pull request by chiwongseekasia**
a new version of footer for hk in english

### CODE REVIEW

File: jobsDB/Footer2/components/FooterLinks/FooterLinks.less [hunk 1]
```diff
@@ -0,0 +1,28 @@
+@import (reference) "~seek-asia-style-guide/theme";
+@import (reference) "../../footer2.less";
+
+@list-line-height: @grid-base * 4.5;
```
Comment (mengtzu) [inline jobsDB/Footer2/components/FooterLinks/FooterLinks.less]: Any time you are multiplying `@grid-base` by anything other than a whole number, you're going off grid :)  If you need an 18px line height, is that offered by one of the text styles?  If it isn't, are you sure you need 18px?

File: jobsDB/Footer2/footer2.less [hunk 2]
```diff
@@ -0,0 +1,79 @@
+@import (reference) "~seek-asia-style-guide/theme";
+@container-height: @grid-base * 75;
+@container-h-padding: @grid-base * 4;
+@wrapper-max-w: @grid-base * 250;
```
Comment (mengtzu) [inline jobsDB/Footer2/footer2.less]: Talk to @patrickliu-sa  about container max widths, this behaviour should be encapsulated into a component you can use instead

File: jobsDB/Footer2/footer2.less [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+@wrapper-up-padding-top: @grid-base * 7.5;
```
Comment (mengtzu) [inline jobsDB/Footer2/footer2.less]: Non-whole multiplier

File: jobsDB/Footer2/footer2.less [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+@wrapper-low-v-padding: @grid-base * 10;
+@title-padding-bottom: @grid-base * 3;
+@footer-font-size: unit(@base-font-size * 1.4, px);
```
Comment (mengtzu) [inline jobsDB/Footer2/footer2.less]: We shouldn't be manually setting font size - can we not use a text style, either via component or mixin?

File: jobsDB/Footer2/footer2.less [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+.container {
+  display: table;
+  cursor: default;
+  height: @container-height;
+  margin: 0 auto;
+  background-color: @saGrey6;
+  padding: 0 @container-h-padding;
+  width: 100%;
+  font-family: @base-font-stack;
```
Comment (mengtzu) [inline jobsDB/Footer2/footer2.less]: Why is this necessary?

File: jobsDB/Footer2/footer2.less (hunk 2 above)
Comment (mengtzu) [reply jobsDB/Footer2/footer2.less]: Also feel free to expand out the `w` and `h` and such, we don't need to save characters.

File: jobsDB/Footer2/footer2.less [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+@container-h-padding: @grid-base * 4;
+@wrapper-max-w: @grid-base * 250;
+@wrapper-up-padding-top: @grid-base * 7.5;
+@wrapper-low-v-padding: @grid-base * 10;
+@title-padding-bottom: @grid-base * 3;
+@footer-font-size: unit(@base-font-size * 1.4, px);
+@copyright-padding-top: @grid-base * 5;
+
+.container {
+  display: table;
```
Comment (mengtzu) [inline jobsDB/Footer2/footer2.less]: Any particular reason why?

File: jobsDB/Footer2/footer2.less [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+  width: 100%;
+  font-family: @base-font-stack;
+  font-weight: @sa-semi-bold;
+}
+
+.wrapper {
+  max-width: @wrapper-max-w;
+  margin: auto;
+  @media @mobile {
+    max-width: @wrapper-max-w / 2.3;
```
Comment (mengtzu) [inline jobsDB/Footer2/footer2.less]: This looks super weird.  Hopefully we could use a standard component to control max width, otherwise I'd assume we'd not need a max width on mobile because there's really not that much width to go around, but if we really do need this bespoke one it'd be better to make it more readable.  Even just using the px value would probably be clearer here.

(do we really need it though?)

File: jobsDB/Footer2/footer2.less [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+  text-decoration: none;
+}
+
+.title {
+  color: @dbBrand;
+  padding-bottom: @title-padding-bottom;
+}
+
+.lower-content {
+  background-color: transparent !important;
```
Comment (mengtzu) [inline jobsDB/Footer2/footer2.less]: `!important` is a code smell :)  Why do we need them here, how are these values getting stomped?

File: jobsDB/Footer2/footer2.less [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+.content:extend(.lower-content) {
+  padding-left: 0 !important;
+  padding-bottom: 0 !important;
+}
+
+.copyright {
+  color: @saGrey2;
+  font-size: unit(@footer-font-size, px);
```
Comment (mengtzu) [inline jobsDB/Footer2/footer2.less]: We shouldn't need to do this, isn't there a text style?

File: react/TwitterIcon/TwitterIcon.js [hunk 10]
```diff
@@ -1,23 +1,26 @@
...
+import svgMarkupPlain from './TwitterIconPlain.svg';
 import svgMarkupFilled from './TwitterIconFilled.svg';
 
 import PropTypes from 'prop-types';
 import React from 'react';
 
 import Icon from '../private/Icon/Icon';
 
-export default function TwitterIcon({ filled, ...props }) {
+export default function TwitterIcon({ filled, plain, ...props }) {
```
Comment (mengtzu) [inline react/TwitterIcon/TwitterIcon.js]: I assume plain is a different presentation of the icon?  Do we actually use both presentations or is one just a SEEK ANZ leftover?

File: react/TwitterIcon/TwitterIcon.js [hunk 11]
```diff
@@ -1,23 +1,26 @@
...
   const markup = filled ? svgMarkupFilled : svgMarkup;
 
-  return <Icon markup={markup} {...props} />;
+  return <Icon markup={plain ? svgMarkupPlain : markup} {...props} />;
```
Comment (mengtzu) [inline react/TwitterIcon/TwitterIcon.js]: I love a good trinary but I think the double trinary is obscuring the relationship between the props.  It'd be good to make the conditionals here more explicit

File: jobsDB/Footer2/footer2.less (hunk 2 above)
Comment (patrickliu-sa) [reply jobsDB/Footer2/footer2.less]: we came up with these breakpoints and corresponding container size:
- screen < 768px, container = 100%
- screen >= 768px, container = 748px
- screen >= 992px, container = 968px
- screen >= 1200px, container = 1168px

File: jobsDB/Footer2/components/Rows/Rows.js [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+
+const renderRows = (el, index) => (
+  <div key={index} className={styles.rows}>{el}</div>
+);
+
+export default function Rows({ children }) {
+  return (
+    <div
+      className={classnames({
+        [styles.rows]: true
```
Comment (mengtzu) [inline jobsDB/Footer2/components/Rows/Rows.js]: You don't need to use classnames if you're only using one, non-conditional class

File: jobsDB/Footer2/localization/en-hk.js [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+          {cfs(<ListItem value={messages['footer2.career']} noShadow compact className={styles.content} descriptionProps={{ semiStrong: true, whistling: true }} />, messages['footer2.careerLink'])}
+          {cms(<ListItem value={messages['footer2.contactUs']} noShadow compact className={styles.content} descriptionProps={{ semiStrong: true, whistling: true }} />, messages['footer2.contactUsLink'])}
+          <DropdownList
+            value={messages['footer2.partners']}
+            noShadow
+            compact
+            className={styles.content}
+            descriptionProps={{ semiStrong: true, whistling: true }}>
+
+            {ext(<ListItem value='BabaJob (India)' noShadow compact className={styles['lower-content']} descriptionProps={{ semiStrong: true, whistling: true }} />, 'http://www.babajob.com/')}
```
Comment (mengtzu) [inline jobsDB/Footer2/localization/en-hk.js]: Looks like we're hardcoding some urls here, getting others from localisation?

File: jobsDB/Footer2/localization/en-hk.js [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+import Rows from '../components/Rows/Rows';
+import { Columns, ListItem, Section, Text, FacebookIcon, TwitterIcon, AppleIcon, AndroidIcon } from 'seek-asia-style-guide/react';
+import DropdownList from '../components/DropdownList/DropdownList';
+import { getLocalization } from '../../localization';
+
+const UpperFooter = ({ linkRenderer, hrefRenderer }) => {
+  const messages = getLocalization({ language: 'en', country: 'hk' });
+
+  const domain = (content, link) => linkRenderer({ href: hrefRenderer({ link }), children: content });
+  const cfs = (content, link) => linkRenderer({ href: hrefRenderer({ link: `${messages['footer2.cfsLink']}${link}` }), children: content });
```
Comment (mengtzu) [inline jobsDB/Footer2/localization/en-hk.js]: Are these link functions repeated from `FooterLinks`?

File: jobsDB/Footer2/Footer2.js [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+  const baseUrl = `https://${country}.${domainUrl}`;
+
+  const messages = getLocalization({ language, country });
+  const upperFooter = getUpperFooter({ language, country });
+
+  return (
+    <footer className={styles.container}>
+      <div className={styles.wrapper}>
+        <div className={styles['upper-wrapper']}>
+          { hrefRenderer ? upperFooter({ linkRenderer, hrefRenderer }) : upperFooter({ linkRenderer, hrefRenderer: makeDefaultHrefRenderer(baseUrl) }) }
```
Comment (mengtzu) [inline jobsDB/Footer2/Footer2.js]: Might be easier to make `makeDefaultHrefRenderer(baseUrl)` the default value of the prop

File: jobsDB/Footer2/components/DropdownList/DropdownList.js [hunk 16]
```diff
@@ -0,0 +1,71 @@
+import styles from './DropdownList.less';
```
Comment (mengtzu) [inline jobsDB/Footer2/components/DropdownList/DropdownList.js]: This component seems hauntingly familiar, did we have something similar already?  `ShowMore`?

File: jobsDB/Footer2/components/DropdownList/DropdownList.less [hunk 17]
```diff
@@ -0,0 +1,33 @@
...
+  cursor: pointer;
+}
+
+.dropdown {
+  display: flex;
+  flex-direction: column;
+  padding-top: @dropdown-padding-top;
+}
+
+.dropdown-no-show:extend(.dropdown) {
```
Comment (mengtzu) [inline jobsDB/Footer2/components/DropdownList/DropdownList.less]: It's less important than normal due to CSS modules, but probably better to standardise on `camelCase` for css class names

File: jobsDB/Footer2/localization/en-hk.js [hunk 18]
```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styles from '../footer2.less';
+import Rows from '../components/Rows/Rows';
+import { Columns, ListItem, Section, Text, FacebookIcon, TwitterIcon, AppleIcon, AndroidIcon } from 'seek-asia-style-guide/react';
+import DropdownList from '../components/DropdownList/DropdownList';
+import { getLocalization } from '../../localization';
+
+const UpperFooter = ({ linkRenderer, hrefRenderer }) => {
```
Comment (mengtzu) [inline jobsDB/Footer2/localization/en-hk.js]: Does it really make sense to have a whole component in a localisation file?  I don't think that really matches expectations.

File: jobsDB/Footer2/components/Rows/Rows.js [hunk 19]
```diff
@@ -0,0 +1,24 @@
...
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import classnames from 'classnames';
+
+const renderRows = (el, index) => (
+  <div key={index} className={styles.rows}>{el}</div>
+);
+
+export default function Rows({ children }) {
```
Comment (mengtzu) [inline jobsDB/Footer2/components/Rows/Rows.js]: I'm a little torn on this component because I love breaking things out, but on the other hand it's doing so little.

Is this in service to a design pattern?  It seems odd to have a top level `<Columns />` component but make `<Rows />` local to a component

File: jobsDB/Footer2/Footer2.js [hunk 20]
```diff
@@ -0,0 +1,75 @@
...
+
+  DefaultLinkRenderer.propTypes = {
+    children: PropTypes.array,
+    href: PropTypes.string
+  };
+
+  return DefaultLinkRenderer;
+};
+
+const makeDefaultHrefRenderer = baseUrl => {
```
Comment (mengtzu) [inline jobsDB/Footer2/Footer2.js]: Is this really a `Renderer`?  Returns a string, not jsx right?

File: jobsDB/Footer2/localization/index.js [hunk 21]
```diff
@@ -0,0 +1,12 @@
...
+const localization = {
+  'en-hk': enHkLocalization
+};
+
+export const getUpperFooter = ({ language, country }) => {
+  return localization[`${language.toLowerCase()}-${country.toLowerCase()}`] ?
+    localization[`${language.toLowerCase()}-${country.toLowerCase()}`] : {};
+};
+
+export default getUpperFooter;
```
Comment (mengtzu) [inline jobsDB/Footer2/localization/index.js]: I'm not sure that's what you'd expect to import from a localisation index

File: react/Columns/Columns.less [hunk 22]
```diff
@@ -20,6 +20,9 @@
     .columns_tight & {
       padding: 0 (@grid-gutter-width / 4);
     }
+    .columns_loose & {
```
Comment (mengtzu) [inline react/Columns/Columns.less]: Is `loose` a design pattern?

File: react/ListItem/ListItem.js [hunk 23]
```diff
@@ -19,12 +19,14 @@ export default function ListItem({
...
   secondLevel,
-  descriptionProps
+  descriptionProps,
+  className
 }) {
   return (
 
     <div
       className={classnames({
+        [className]: className,
```
Comment (mengtzu) [inline react/ListItem/ListItem.js]: Probably should be exercised in a test

File: react/Columns/Columns.js [hunk 24]
```diff
@@ -8,12 +8,13 @@ const renderColumn = (el, index) => (
...
 );
 
-export default function Columns({ children, tight, flexible }) {
+export default function Columns({ children, tight, loose, flexible }) {
   return (
     <div
       className={classnames({
         [styles.columns]: true,
         [styles.columns_tight]: tight,
+        [styles.columns_loose]: loose,
```
Comment (mengtzu) [inline react/Columns/Columns.js]: Needs a test case

File: jobsDB/Footer2/components/Rows/Rows.js (hunk 19 above)
Comment (mengtzu) [reply jobsDB/Footer2/components/Rows/Rows.js]: Also if this is a component it should have tests :)

File: jobsDB/Footer2/components/FooterLinks/FooterLinks.js [hunk 25]
```diff
@@ -0,0 +1,31 @@
...
+
+import styles from './FooterLinks.less';
+
+const FooterLinks = ({ messages, linkRenderer, hrefRenderer }) => {
+  const cfs = (content, link) => linkRenderer({ href: hrefRenderer({ link: `${messages['footer2.cfsLink']}${link}` }), children: content, className: styles.link });
+  const cms = (content, link) => linkRenderer({ href: hrefRenderer({ link: `${messages['footer2.cmsLink']}${link}` }), children: content, className: styles.link });
+
+  return (
+    <ul className={styles.list}>
+      <li className={styles.item}>
```
Comment (mengtzu) [inline jobsDB/Footer2/components/FooterLinks/FooterLinks.js]: Is our `<ListItem />` component not suitable here?

File: jobsDB/Footer2/components/DropdownList/DropdownList.js (hunk 16 above)
Comment (mengtzu) [reply jobsDB/Footer2/components/DropdownList/DropdownList.js]: Also would need tests!

File: jobsDB/Footer2/Footer2.js [hunk 26]
```diff
@@ -0,0 +1,75 @@
...
+  };
+
+  DefaultHrefRenderer.PropTypes = {
+    link: PropTypes.string.isRequired
+  };
+
+  return DefaultHrefRenderer;
+};
+
+const Footer2 = ({ language, country, domainUrl, linkRenderer, hrefRenderer }) => {
```
Comment (mengtzu) [inline jobsDB/Footer2/Footer2.js]: Component will need tests

File: react/TwitterIcon/TwitterIcon.js (hunk 11 above)
Comment (mengtzu) [reply react/TwitterIcon/TwitterIcon.js]: Also I think there's actually enough logic in here that it'll need a test, unlike most icon files 🤔

**Comment by mengtzu** [comment]
It would be great if we could call this something other than `Footer2` 😂 

If it legitimately is HK only for now, might be best to make it a subcomponent of the existing footer, and return this component if the hk country code prop is passed in.  That way consumers don't have to have to reference a temporary component name or make the decision themselves.

File: jobsDB/Footer2/footer2.less (hunk 2 above)
Comment (mengtzu) [reply jobsDB/Footer2/footer2.less]: We'll want to expose those as variables, but also have `<PageBlock />` implement them so hopefully nobody needs to actually use the variables 😂

File: jobsDB/Footer2/components/FooterLinks/FooterLinks.less (hunk 1 above)
Comment (patrickliu-sa) [reply jobsDB/Footer2/components/FooterLinks/FooterLinks.less]: I understand the intent of using `line-height`, which's not a multiplier of `@grid-base`. Some of them are hardly tie to the defined grid according to current typography setting.
https://brand.seekasia.com/product/typography/
Probably need to compensated by `vertical-align` with top/bottom padding to make component fit into the grid.

File: jobsDB/Footer2/Footer2.js (hunk 15 above)
Comment (CedricPoon-SeekAsia) [reply jobsDB/Footer2/Footer2.js]: @chiwongseekasia will follow up

File: jobsDB/Footer2/components/DropdownList/DropdownList.js (hunk 16 above)
Comment (CedricPoon-SeekAsia) [reply jobsDB/Footer2/components/DropdownList/DropdownList.js]: @chiwongseekasia 
this is Chi, yup it's familiar but we base on our design it is expand from top not like the 'show more component expand on bottom.

File: react/Columns/Columns.less (hunk 22 above)
Comment (CedricPoon-SeekAsia) [reply react/Columns/Columns.less]: @chiwongseekasia: will follow up

File: jobsDB/Footer2/components/FooterLinks/FooterLinks.js (hunk 25 above)
Comment (CedricPoon-SeekAsia) [reply jobsDB/Footer2/components/FooterLinks/FooterLinks.js]: @chiwongseekasia:
yo Chi again, I think the list item is not really suit in here, cause the padding we can't have a props to change it show I will not suggest using list item here.

File: jobsDB/Footer2/footer2.less (hunk 3 above)
Comment (patrickliu-sa) [reply jobsDB/Footer2/footer2.less]: For block element we can tolerate more vertical space alignment as it will get the tallest child element as the height reference :)

Comment (mengtzu) [reply jobsDB/Footer2/footer2.less]: Even if it works somehow, we shouldn't just use a non-whole multiplier, because it *looks broken*.  Someone will come along and refactor it.

If we need an unusual dimension we should probably do something that expresses *why* something is a certain way.  Like say you had to offset a margin due to a border taking up some of the grid space:

```
border: 1px solid @saLuridPurple;
margin-top: (@grid-base * 3) - 1;  // border offset
```

Not only is it obvious why we're doing this and that it's OK, the relationship doesn't change based on the value of `@grid-base`.  Whereas if we expressed that as `@grid-base * 2.75` it's going to get weird if we change @grid-base to `6px` (same as SEEK ANZ) or `5px` (because we decide it's easier to reason about) or whatever (`13px` because prime numbers are cool in 2023?)

File: jobsDB/Footer2/components/DropdownList/DropdownList.js (hunk 16 above)
Comment (mengtzu) [reply jobsDB/Footer2/components/DropdownList/DropdownList.js]: @chiwongseekasia feels like that's a candidate for a prop on `ShowMore` then.   I don't want to overcomplicate components but if there's a lot of repeated code it feels like the lesser of two evils.

File: react/TwitterIcon/TwitterIcon.js (hunk 10 above)
Comment (chiwongseekasia) [reply react/TwitterIcon/TwitterIcon.js]: @patrickliu-sa are we need to keep both?

**Comment by chiwongseekasia** [comment]
@mengtzu I am thinking the final goal is using ver 2 to replace ver 1.
how u think we name it as 'hk footer' first?

**Comment by mengtzu** [comment]
`HKFooter` would be fine.  Move it inside the existing footer, return it instead of the existing footer if the country prop is `hk`, eventually just replace it entirely.  Consumers don't need to make any changes that way.

File: react/TwitterIcon/TwitterIcon.js (hunk 10 above)
Comment (patrickliu-sa) [reply react/TwitterIcon/TwitterIcon.js]: Seems on SEEK ANZ kept the two versions.  We just need the one without circled

File: jobsDB/Footer2/footer2.less (hunk 4 above)
Comment (CedricPoon-SeekAsia) [reply jobsDB/Footer2/footer2.less]: Agree and will use ```<Text whistling></Text>``` instead

File: jobsDB/Footer2/footer2.less (hunk 5 above)
Comment (CedricPoon-SeekAsia) [reply jobsDB/Footer2/footer2.less]: 1. ```width``` is used for spreading the footer to whole page horizontally
2. ```padding``` is used for padding while resizing responsively
3. ```background-color``` is governing the footer color to align with the design

File: jobsDB/Footer2/footer2.less (hunk 8 above)
Comment (CedricPoon-SeekAsia) [reply jobsDB/Footer2/footer2.less]: All ```!important``` will be removed

File: jobsDB/Footer2/footer2.less (hunk 9 above)
Comment (CedricPoon-SeekAsia) [reply jobsDB/Footer2/footer2.less]: Will be adopting ```<Text></Text>``` instead

File: react/TwitterIcon/TwitterIcon.js (hunk 10 above)
Comment (CedricPoon-SeekAsia) [reply react/TwitterIcon/TwitterIcon.js]: Discussed with @patrickliu-sa, we are changing ```plain``` to ```generic```

File: jobsDB/Footer2/footer2.less (hunk 6 above)
Comment (CedricPoon-SeekAsia) [reply jobsDB/Footer2/footer2.less]: Changing the behaviour of ```.container``` to avoid ```display: table;```

**Comment by CedricPoon-SeekAsia** [comment]
WIP ;-)
The followings are pending to be done.
1. Add ```makeDefaultHrefLink()``` the default props
2. Rename ```footer2``` to ```HKFooter```
3. Embed ```localization``` under component

**Comment by CedricPoon-SeekAsia** [comment]
@mengtzu @chiwongseekasia @patrickliu-sa should be available for another round of review, pls feel free to drop comments :-)

File: jobsDB/Footer/Footer.js [hunk 27]
```diff
@@ -3,12 +3,16 @@ import styles from './footer.less';
...
 import FooterLinks from './components/FooterLinks/FooterLinks';
+import FooterHK from './FooterHK/FooterHK';
 import localization from './localization';
 
 const Footer = ({ language, country }) => {
   const year = new Date().getFullYear();
   const messages = localization[`${language}-${country}`];
 
+  if (country === 'hk') {
+    return (<FooterHK />);
```
Comment (mengtzu) [inline jobsDB/Footer/Footer.js]: do you need to pass props through here?

File: jobsDB/Footer/FooterHK/components/DropdownList/DropdownList.js [hunk 28]
```diff
@@ -0,0 +1,65 @@
+import styles from './DropdownList.less';
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { Text, ChevronIcon } from 'seek-asia-style-guide/react';
+
+export default class DropdownList extends Component {
```
Comment (mengtzu) [inline jobsDB/Footer/FooterHK/components/DropdownList/DropdownList.js]: Component name still seems a little generic, is there something you can call it that expresses why it's a footer subcomponent and not up in `/react`?

File: jobsDB/Footer/FooterHK/FooterHK.js [hunk 29]
```diff
@@ -0,0 +1,268 @@
+import React from 'react';
+import styles from './footerHK.less';
```
Comment (mengtzu) [inline jobsDB/Footer/FooterHK/FooterHK.js]: Capitalise this - `FooterHK.less`

File: jobsDB/Footer/FooterHK/FooterHK.js [hunk 30]
```diff
@@ -0,0 +1,268 @@
...
+  const cfsLink = (content, link) => linkRenderer({ href: hrefLink({ link: `${messages['FooterHK.cfsLink']}${link}` }), children: content });
+  const wordpressLink = (content, link) => linkRenderer({ href: hrefLink({ link: `${messages['FooterHK.cmsLink']}${link}` }), children: content });
+  const externalLink = (content, link) => linkRenderer({ href: link, children: content });
+
+  return (
+    <footer className={styles.container}>
+      <PageBlock>
+        <div className={styles.upperWrapper}>
+          <Columns>
+            <div className={styles.columnGroup}>
```
Comment (mengtzu) [inline jobsDB/Footer/FooterHK/FooterHK.js]: Why nesting the same kind of div? 😕

File: jobsDB/Footer/Footer.js (hunk 27 above)
Comment (CedricPoon-SeekAsia) [reply jobsDB/Footer/Footer.js]: @mengtzu in this place 'hk' is already a default prop inside ```<FooterHK />```. Thus no additional props needed to be passed

File: jobsDB/Footer/FooterHK/FooterHK.js (hunk 29 above)
Comment (CedricPoon-SeekAsia) [reply jobsDB/Footer/FooterHK/FooterHK.js]: @mengtzu sure

File: jobsDB/Footer/Footer.js (hunk 27 above)
Comment (CedricPoon-SeekAsia) [reply jobsDB/Footer/Footer.js]: As advised by @chiwongseekasia, will pass ```country``` and ```language``` to ```<FooterHK />```

Comment (mengtzu) [reply jobsDB/Footer/Footer.js]: That's not the only prop - how about `LinkRenderer`?

Comment (CedricPoon-SeekAsia) [reply jobsDB/Footer/Footer.js]: @mengtzu May I ask can I refer to ```descriptionProps``` in ListItem? It supports extensibility though.

File: jobsDB/Footer/FooterHK/components/DropdownList/DropdownList.js (hunk 28 above)
Comment (chiwongseekasia) [reply jobsDB/Footer/FooterHK/components/DropdownList/DropdownList.js]: @mengtzu @patrickliu-sa 
Guys I think Cedric wrote a useful dropdown component, which can wrap any item in it.
I think we should move out to an independent component rather then hide it in footer.

File: jobsDB/Footer/FooterHK/FooterHK.js (hunk 30 above)
Comment (CedricPoon-SeekAsia) [reply jobsDB/Footer/FooterHK/FooterHK.js]: @mengtzu cos it is just kind of mimic on the behaviour of ```<Columns />```. Can be further refined to single ```<div>``` if needed.

Comment (mengtzu) [reply jobsDB/Footer/FooterHK/FooterHK.js]: Why would you need to mimic `<Columns/>` when you're in `<Columns />`?

That's really confusing!

File: jobsDB/Footer/Footer.js (hunk 27 above)
Comment (mengtzu) [reply jobsDB/Footer/Footer.js]: If you mean what I think you mean, that pattern is for specifying a set of props for a subcomponent (it's also not great, there's just a few places we have to do it).

In this case while we've organised the thing as a subcomponent, it's really impersonating the parent component, replacing it completely.  It should get all the props.

File: jobsDB/Footer/Footer.js [hunk 31]
```diff
@@ -3,12 +3,16 @@ import styles from './footer.less';
...
+import FooterHK from './FooterHK/FooterHK';
 import localization from './localization';
 
-const Footer = ({ language, country }) => {
+const Footer = ({ language, country, localeProps }) => {
   const year = new Date().getFullYear();
   const messages = localization[`${language}-${country}`];
 
+  if (country === 'hk') {
+    return (<FooterHK {...localeProps} language={language} country={country} />);
```
Comment (mengtzu) [inline jobsDB/Footer/Footer.js]: Still not passing through link rederers right?  Will need those to be able to do any omniture tracking

File: jobsDB/Footer/FooterHK/components/DropdownList/DropdownList.js (hunk 28 above)
Comment (CedricPoon-SeekAsia) [reply jobsDB/Footer/FooterHK/components/DropdownList/DropdownList.js]: This component will be separated as an individual SEEK Asia Component soon after closing of this PR.
